### PR TITLE
Unify types/values passed to upsertRoom/updateRoom in admin components

### PR DESCRIPTION
**Pull request by yurii-lubynets**
connected to https://github.com/sparkletown/sparkle/pull/1330 (see review comments in https://github.com/sparkletown/sparkle/pull/1330#pullrequestreview-637305519)

### CODE REVIEW

File: src/pages/Admin/Venue/Details/VenueDetails.tsx [hunk 1]
```diff
@@ -153,12 +153,12 @@ const VenueDetails: React.FC<VenueDetailsProps> = ({ venue }) => {
...
 
   const handleEditRoomSave = async (values: RoomData_v2, index: number) => {
-    const newData: RoomData_v2 = {
+    const roomValues: RoomData_v2 = {
       ...editingRoom,
       ...values,
     };
 
-    await updateRoom(newData, venueId!, user, index);
+    await updateRoom(roomValues, venueId!, user, index);
```
Comment (0xdevalias) [inline src/pages/Admin/Venue/Details/VenueDetails.tsx:161]: We want to remove this TypeScript `!` from the codebase as it basically disables type checking.

```suggestion
    await updateRoom(roomValues, venueId, user, index);
```

File: src/pages/Admin/Venue/Rooms/RoomsForm.tsx [hunk 2]
```diff
@@ -101,7 +101,7 @@ interface RoomInnerForm {
   editingRoomIndex?: number;
 }
 
-export type FormValues = Partial<Yup.InferType<typeof validationSchema>>;
+export type FormValues = Yup.InferType<typeof validationSchema>;
```
Comment (0xdevalias) [inline src/pages/Admin/Venue/Rooms/RoomsForm.tsx:104]: Why did we change the type from using `Partial` here? Is that correct? Will it break anything? Are the form values always guaranteed to be set in the `onSubmit`? Or does this just end up making our types 'lie' about the underlying data at 'compile time', which will then lead to hard to debug/identify runtime issues.

**Review by 0xdevalias — COMMENTED** [review]
Comments/Questions/etc

File: src/pages/Admin/Venue/Rooms/RoomsForm.tsx (hunk 2 above)
Comment (yurii-lubynets) [reply src/pages/Admin/Venue/Rooms/RoomsForm.tsx:104]: All fields in the `RoomsForm` are required and there is no sense to make them partial.
And it fixed the type checking `onSubmit` with `RoomInput`. Guess now it's more secure

Comment (0xdevalias) [reply src/pages/Admin/Venue/Rooms/RoomsForm.tsx:104]: So long as everything still works as expected if I only partially fill out the form, then sounds good.

**Review by 0xdevalias — APPROVED** [review]
[CRO] Code review looks good, I haven't clicked through/tested this, so if we're sure it works/won't break things (and https://github.com/sparkletown/sparkle/pull/1338#discussion_r617090844 is satisfied), then LGTM!

**Pull request merged**